gui/default/syncthing/device/shareDeviceDialogView.html

```diff
@@ -0,0 +1,52 @@
+<modal id="share-device-confirmation" status="warning" icon="{{shareDeviceParams.icon}}" heading="{{shareDeviceParams.heading}}" large="no" closeable="yes">
+  <div class="modal-body">
+    <p>
+      <span translate>The following content will automatically be inserted into a message.</span>
+      <span ng-switch="shareDeviceParams.type">
+        <span ng-switch-when="email">
+          <span translate>You should then use your own email account to send it.</span>
+          <span translate>Proceed only if you have a registered email client.</span>
```

Comment:
The `registered email client` is a Microsoft term. Windows literally tells you that if there's no registered handler for the email protocol. I assumed anyone anyone who'd ever used Windows in the last 30 years would be familiar with the term, but I'll try to make the sentence more accessible nevertheless 😉.

I'm not a fan of using the word `app` or `program` here though, because nowadays quite a lot of people have their Web browser (or rather Gmail specifically) registered as the default e-mail handler, which isn't exactly a separate application.

> What else should the user do?

If the "share" button doesn't work, the user is supposed to copy the content to clipboard, and then manually proceed to go to their e-mail and send it there. I think some kind of a warning is necessary here, because the OS/browser will often go wonky if there's no registered application and display some kind of a warning with a lot of technical jargon, which the user have no idea what to do with.

Reply:
I agree that "program" or "application" is not well suited and too technical. But the term "app" has evolved into a widely known substitute for the "thing on a screen you interact with". Even Gmail in the browser can be called an app, and Syncthing's own GUI is another prime example.

If the browser gives a stupid error message, that's their fault and we try to set the right expectations beforehand. That's what I like about the new dialog. When the user sees unexpected or cryptic messages, it's very clearly like "OK this way doesn't work, let's go back one step and try differently". At that point anyone having mastered Syncthing to the point of recommending it to others should know basic things like how to forward a text displayed on screen to a friend via their usual messenger of choice.

My "what else should they do" comment was specifically targeted at the "You should..." sentence, which I found to just repeat what the user is already trying to do. The "proceed only..." warning sounds scary including the registration part, and leaving no recommendation for those who would rather not do anything like registering. So just tell them what should happen, let them see if it works (because we cannot reliably predict) and describe the alternative. Hence my suggested wording.